src/node_worker.cc

```diff
@@ -132,7 +132,17 @@ class WorkerThreadData {
  public:
   explicit WorkerThreadData(Worker* w)
     : w_(w) {
-    CHECK_EQ(uv_loop_init(&loop_), 0);
+    if (int ret = uv_loop_init(&loop_) != 0) {
+      char err_buf[128];
+      char error_str[256];
+      uv_err_name_r(ret, err_buf, sizeof(err_buf));
+      snprintf(error_str,
+        sizeof(err_buf) + sizeof("ERR_WORKER_INIT_FAILED: ") + 1,
+        "ERR_WORKER_INIT_FAILED: %s",
+        err_buf);
+      w->custom_error_ = error_str;
```

Comment:
I think you may need to switch `custom_error_` to be an `std::string` in order for this to work, otherwise it’s a dangling pointer to `error_str` afterwards.

Reply:
@addaleax, Thanks. Changed to `std::string`.